docs/developers/celestia-app-vesting.md

```diff
@@ -0,0 +1,404 @@
+---
+sidebar_label: How to create a vesting account with Celestia App
+description: Learn how to generate a vesting account using Celestia App.
+---
+
+# How to create a vesting account with Celestia App
+<!-- markdownlint-disable MD013 -->
+
+In this guide, we will learn how to create a vesting account using
+celestia-appd for both a local devnet and on Mocha testnet.
+
+## Local devnet
+
+First, download and [install Celestia App](../../nodes/celestia-app),
+selecting the
+[network and corresponding version](../../nodes/participate)
+that you would like to use.
+
+### Setting up the local devnet
+
+#### Run the devnet
+
+Next, change into the `$HOME/celestia-app` directory and
+run the single-node-devnet script.
+
+```bash
+cd $HOME/celestia-app
+./scripts/build-run-single-node.sh
```

Comment:
[question][discussion] I don't think @celestiaorg/celestia-core maintainers treat scripts located in `./scripts/` as public API. I don't expect any breaking changes to this script in the short term but just flagging that we should probably be careful to not introduce breaking changes to any of the scripts that are referenced from the docs.

Reply:
good point, i'm not sure here actually. do you think it is a good idea to just put a script in to the docs for people to copy on their own?

Reply:
hmm if we put the script into the docs then it may become stale / break due to changes in celestia-app. I think I'd rather the script live in the celestia-app repo.

If we really want to ensure the docs don't break, we could introduce a CI test that tries to run this script. Or alternatively, we could write a new script that contains the steps on this docs page that runs in CI so we don't break it accidentally. Thoughts?

Reply:
if it's worth the testing resources, I think it would be worth it. at the moment though, not sure where it lies in terms of scope/bandwidth. so will leave this PR open for now and continue to think on it

Reply:
i noticed that the single node script is in the readme. is there ci for that there? maybe it is possible to add a comment in the readme that says "TODO: if you delete this, update docs too" by the node script?

Reply:
Unfortunately there isn't CI that ensures our README commands run as expected. I can add a TODO in celestia-app repo if we want to merge this as-is

Reply:
Created https://github.com/celestiaorg/celestia-app/issues/2007 as a potential longer term solution

Reply:
If the scripts are needed to run as an end-user, then they're part of the app's public-facing API and should be 1) documented and 2) tested in CI.